src/domain/application/components/ApplicationSendDialog.tsx

```diff
@@ -0,0 +1,184 @@
+      orderPaperDecision: false,
+      paperDecisionReceiver: null,
+    },
+  });
+
+  const { handleSubmit, formState, register, watch, setValue, resetField, reset } = formContext;
+  const [orderPaperDecisionChecked] = watch(['orderPaperDecision']);
+  const isConfirmButtonEnabled = formState.isValid;
+  const dialogTitle = t('hakemus:sendDialog:title');
+
+  useEffect(() => {
+    formContext.setValue('applicationId', applicationId);
+  }, [formContext, applicationId, setValue]);
+
+  async function submitForm(data: ApplicationSendData) {
+    const paperDecisionReceiver = data.orderPaperDecision
+      ? (data.paperDecisionReceiver as PaperDecisionReceiver)
+      : null;
+    onSend(paperDecisionReceiver);
+    onClose();
+  }
+
+  function handleOrderPaperDecisionChange() {
+    const newValue = !orderPaperDecisionChecked;
+    setValue('orderPaperDecision', newValue, {
+      shouldDirty: true,
+    });
+    if (!newValue) {
+      resetField('paperDecisionReceiver', { defaultValue: null });
+    }
+  }
+
+  function handleClose() {
+    reset();
+    onClose();
+  }
+
+  return (
+    <Dialog
+      id="application-send"
+      isOpen={isOpen}
+      aria-labelledby={dialogTitle}
+      variant="primary"
+      close={onClose}
+      closeButtonLabelText={t('common:ariaLabels:closeButtonLabelText')}
+    >
+      <Dialog.Header
+        id="application-send-title"
+        title={dialogTitle}
+        iconLeft={<IconQuestionCircle aria-hidden="true" />}
+      />
+      <FormProvider {...formContext}>
+        <form onSubmit={handleSubmit(submitForm)}>
+          <Dialog.Content>
+            <VisuallyHiddenInput
+              name="applicationId"
+              value={applicationId}
+              aria-hidden="true"
+              readOnly
+            />
+            <Text tag="p" spacingBottom="s">
+              {t('hakemus:sendDialog:instructions')}
+            </Text>
+            <Box marginLeft="var(--spacing-xs)" marginBottom="var(--spacing-s)">
+              <ToggleButton
+                {...register('orderPaperDecision')}
+                id="orderPaperDecision"
+                label={t('hakemus:sendDialog:orderPaperDecision')}
+                checked={orderPaperDecisionChecked}
+                onChange={handleOrderPaperDecisionChange}
+              />
+            </Box>
+            {orderPaperDecisionChecked && (
+              <Box>
+                <Notification label={t('hakemus:sendDialog:paperDecisionNotificationLabel')}>
+                  {t('hakemus:sendDialog:paperDecisionNotificationText')}
+                </Notification>
+                <Grid
+                  templateRows="repeat(3, 1fr)"
+                  templateColumns={{ sm: 'repeat(3, 1fr)', xs: 'repeat(1, 1fr)' }}
+                  marginTop="var(--spacing-s)"
+                  columnGap={4}
+                  rowGap={3}
+                >
+                  <GridItem colSpan={2}>
+                    <TextInput
```

Comment:
Maybe the wrapper component `src/common/components/textInput/TextInput.tsx` could be used here instead.

Reply:
Yes, thank you for pointing this out. Changed it.